modules/dnn/src/layers/scale_layer.cpp

```diff
@@ -78,64 +78,95 @@ class ScaleLayerImpl CV_FINAL : public ScaleLayer
         Mat &outBlob = outputs[0];
         // There is a mode when we multiply a first blob by a second one
         // instead of trainable weights.
-        Mat weights = hasWeights ? (blobs.empty() ? inputs[1] : blobs[0]).reshape(1, 1) : Mat();;
+        Mat weights = hasWeights ? (blobs.empty() ? inputs[1] : blobs[0]).reshape(1, 1) : Mat();
         Mat bias = hasBias ? (blobs.empty() ? inputs[1] : blobs.back()).reshape(1, 1) : Mat();
 
         MatShape inpShape = shape(inpBlob);
+        const int numSamples = inpShape[0];
+        const int numWeightsSamples = (hasWeights || hasBias) && blobs.empty() ? numSamples : 1;
```

Comment:
Here, I assumed that if weights are non-constants, then batch size could be N for input NCHW. i.e. for each sample weight is different ( example- multiplication of two inputs ) and if weights are constants then batch size is equal to 1, assuming it will be same for all samples.